pkg/ingester/ingester.go

```diff
@@ -907,42 +910,42 @@ func (i *Ingester) pushSamplesToAppender(userID string, timeseries []mimirpb.Pre
 		case storage.ErrOutOfBounds:
 			stats.sampleOutOfBoundsCount++
 			updateFirstPartial(func() error {
-				return newIngestErrSampleTimestampTooOld(model.Time(timestamp), labels)
+				return i.limiter.sampler.WrapError(newIngestErrSampleTimestampTooOld(model.Time(timestamp), labels))
 			})
 			return true
 
 		case storage.ErrOutOfOrderSample:
 			stats.sampleOutOfOrderCount++
 			updateFirstPartial(func() error {
-				return newIngestErrSampleOutOfOrder(model.Time(timestamp), labels)
+				return i.limiter.sampler.WrapError(newIngestErrSampleOutOfOrder(model.Time(timestamp), labels))
```

Comment:
This is what I was talking about on the previous PR when I mentioned that we should have different samplers.

With a single sampler, there's no way to tell how many of each errors are happening right? There might be 27 "too old" errors and 3 "out of order" errors, and we could still only see the "out of order" ones.

Reply:
That is very difficult to arrange.  In real life the different kinds of errors will be randomly spaced and over time the sampler will provide a statistical picture of them.

Reply:
I don't think that having a separate sampler for each error would be that difficult to arrange.

But with current approach I'd rather just drop and not log these errors at all, as I don't think it adds much value to see some logs lines from time to time without even being able to tell even how many of each one are happening.

Reply:
This is absolutely not my experience.  I added sampled ["events"](https://github.com/cortexproject/cortex/blob/6c41a643d5556f05dff279e6a66700fae3456aa9/pkg/util/events.go) to Cortex and used them every week to investigate cardinality and other issues.  (removed in #766 because it was never added to blocks)

Reply:
BTW in case it's unclear when I said "That is very difficult to arrange", I mean to get the 27 errors and the 3 errors to come out exactly on a pattern such that you only see the 3.

Reply:
Quick update on this: to be clear, I'm not blocking, because I also wouldn't complain if we completely removed the errors. I think this is the typical thing where Murphy's Law will play on me, but I also have to trust your experience and if you think it's useful, then it probably is.

Reply:
I have updated the PR by creating a different sampler for each new error type